lib/features/modeling/behavior/SubProcessPlaneBehavior.js

```diff
@@ -95,6 +95,41 @@ export default function SubProcessPlaneBehavior(
   }, true);
 
 
+  this.preExecuted('shape.delete', function(context) {
+    var shape = context.shape;
+    if (!isCollapsedSubProcess(shape)) {
+      return;
+    }
+
+    var attachedRoot = elementRegistry.get(planeId(shape));
+
+    if (!attachedRoot) {
+      return;
+    }
+
+    modeling.removeElements(attachedRoot.children.slice());
```

Comment:
The current way we implemented `removeRoot`, it is not a modeling command, but an atomic operation called during `executed/reverted`. It also does not remove any child shapes, but only the root element from the canvas as well as the reference to the diDiagram.

We could create 2 new modeling commands handlers (addRoot/removeRoot) and call them in `postExecuted` instead of having separate `executed/reverted` handlers. Is this what you mean?

Reply:
Exactly. I acknowledge that I missed this with the `diagram-js@8` release. It would be the cannonical way of supporting the multi-root feature end to end though.

Reply:
Thanks for clarifying. I'll look into it

Reply:
Also happy to keep this as a future improvement. Your call!

Reply:
Let's do it as an improvement. This allows us to make progress on other PRs (copy/paste) independently of the changes here